### PR TITLE
Implement allowBackgroundLocationUpdates iOS setting

### PR DESCRIPTION
**Pull request by landyrev**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Feature

### :arrow_heading_down: What is the current behavior?
Currently geolocator sets `allowBackgroundLocationUpdates` equal to `true` by default when the `UIBackgroundModes` contains `location` value inside the app's Info.plist, which leads to `Geolocator.getPositionStream` receiving background geolocations even if you don't need them

### :new: What is the new behavior (if this is a feature change)?
User can disable background geolocations updates even if their app contains `location` value inside the `UIBackgroundModes` key

### :boom: Does this PR introduce a breaking change?
No, `allowBackgroundLocationUpdates` is set to true by default

### :bug: Recommendations for testing
Can be tested on iOS devices and simulators
1. Add `location` value to the `UIBackgroundModes` Info.plist key
2. Request position stream with `getPositionStream` with `locationSettings` equal to:
```dart
AppleSettings(
  allowBackgroundLocationUpdates: false
)
```
3. Send app to the background 
4. Check that there is no background geolocation indicator visible in status bar

### :thinking: Checklist before submitting

- [x] I made sure all projects build.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy](https://dart.dev/tools/pub/versioning).
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I followed the style guide lines ([code style guide](https://github.com/Baseflow/flutter-geolocator/blob/master/CONTRIBUTING.md)).
- [x] I updated the relevant documentation.
- [x] I rebased onto current `master`.

### CODE REVIEW

File: geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m [hunk 1]
```diff
@@ -105,7 +109,8 @@ - (void)startUpdatingLocationWithDesiredAccuracy:(CLLocationAccuracy)desiredAccu
   
 #if TARGET_OS_IOS
   if (@available(iOS 9.0, macOS 11.0, *)) {
-    locationManager.allowsBackgroundLocationUpdates = [GeolocationHandler shouldEnableBackgroundLocationUpdates];
+    locationManager.allowsBackgroundLocationUpdates = allowBackgroundLocationUpdates
+      && [GeolocationHandler shouldEnableBackgroundLocationUpdates];
```
Comment (mvanbeusekom) [inline geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m:113]: There should be a unit test to verify the `allowsBackgroundLocationUpdates` property is set correctly (only when both values are true.

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Thank you for contributing, this looks very good. I have just one request to add an additional test.

File: geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m (hunk 1 above)
Comment (landyrev) [reply geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m:113]: @mvanbeusekom good point, thanks! Added a couple of tests for that

**Review by mvanbeusekom — APPROVED** [review]
LGTM, thanks for this great addition.

**Comment by Faiizii** [comment]
I think it should be in readme file. I implemented the background location updates in one of my app. In the recent updates it stopped working with the following code 
```
Geolocator.getPositionStream(
          locationSettings: LocationSettings(
              accuracy: LocationAccuracy.bestForNavigation,
          )
      )
```

I have ignored the rest of the params for testing purpose. 

When I replaced the location settings with Apple Setting for iOS as suggested above, background location updates are working.

```
AppleSettings(
            accuracy: LocationAccuracy.bestForNavigation,
            allowBackgroundLocationUpdates: true,
            showBackgroundLocationIndicator: true,
            activityType: ActivityType.automotiveNavigation
        );
``` 

I have followed all the steps like allowing location updates in background mode, always location permission.

**Comment by landyrev** [comment]
hey, @Faiizii, what's your `geolocator_apple` version? (you can find it in the `pubspec.lock`) 

Also, do you use the `getCurrentPosition` somewhere in your code? If yes, then it might be related to this issue: https://github.com/Baseflow/flutter-geolocator/pull/1169

**Comment by Faiizii** [comment]
`geolocator_apple` is 2.2.3. No, I've just used the `getPositionStream` to get continuous updates.

**Comment by mvanbeusekom** [comment]
Hi @landyrev, 

Reading the original comment from @Faiizii, it seems that the only problem is that the documentation doesn't explain to use the `AppleSettings` and specifically set the `allowBackgroundLocationUpdates` parameter to `true` if you'd want to enable background support for iOS.

He originally used the `LocationSettings` object which is mentioned in the documentation but doesn't turn on support for background services (as this is Apple specific).

**Comment by landyrev** [comment]
@mvanbeusekom, ah, right. We set allowBackgroundLocationUpdates equal to `true` only when you use `AppleSettings`.

Yeah, fair enough, I'll draft a PR

**Pull request merged**